lib/controllers/root-controller.js

```diff
@@ -800,3 +742,104 @@ export default class RootController extends React.Component {
     });
   }
 }
+
+class TabTracker {
+  constructor(component, name) {
+    this.component = component;
+    this.workspace = component.props.workspace;
+    this.name = name;
+
+    this.stateKey = `${name}TabActive`;
+    this.dockItemKey = `${name}DockItem`;
+    this.controllerKey = `${name}TabController`;
+  }
+
+  getState() {
+    return this.component.state[this.stateKey];
+  }
+
+  setStateKey(value) {
+    return new Promise(resolve => {
+      this.component.setState({[this.stateKey]: value}, resolve);
```

Comment:
It turns out that I could refactor TabTracker to have injected functions instead of reaching into Component directly without too much trouble. Thanks 🙇